test/azlinux_test.go

```diff
@@ -325,10 +325,40 @@ index 0000000..5260cb1
 +
 +echo "Added another new file"
 `)
-		src2Patch3Context := llb.Scratch().File(
-			llb.Mkfile(src2Patch3File, 0o600, src2Patch3Content),
-		)
-		src2Patch3ContextName := "patch-context"
+
+		src2Patch4Content := []byte(`
+diff --git a/file4 b/file4
+new file mode 100700
+index 0000000..5260cb1
+--- /dev/null
++++ b/file4
+@@ -0,0 +1,3 @@
++#!/usr/bin/env bash
++
++echo "Added yet another new file"
+`)
+
+		src2Patch5Content := []byte(`
+diff --git a/file5 b/file5
+new file mode 100700
+index 0000000..5260cb1
+--- /dev/null
++++ b/file5
+@@ -0,0 +1,3 @@
++#!/usr/bin/env bash
++
++echo "Added yet again...another new file"
+`)
+
+		const src2Patch4File = "patches/patch4"
+		const src2Patch5File = "patches/patch5"
+		const patchContextName = "patch-context"
+
+		patchContext := llb.Scratch().
+			File(llb.Mkfile(src2Patch3File, 0o600, src2Patch3Content)).
+			File(llb.Mkdir("patches", 0o755)).
+			File(llb.Mkfile(src2Patch4File, 0o600, src2Patch4Content)).
+			File(llb.Mkfile(src2Patch5File, 0o600, src2Patch5Content))
```

Comment:
question, so here you are only building the llb definition that's why the file action's order does not matter? I am referring to seeing here first creating the file `src2Patch3File` and then the directory where this lives `patches`

Reply:
`src2Patch3` is not in `patches`, it is at the root of the context.
Order actually does matter here in that you can't create a file in a subpath that does not exist.